react-quiz/src/API.ts

```diff
@@ -0,0 +1,27 @@
+import { shuffleArray } from './utils';
+
+export type Question = {
+  category: string;
+  correct_answer: string;
+  difficulty: string;
+  incorrect_answers: string[];
+  question: string;
+  type: string;
+};
+
+export enum Difficulty {
+  EASY = "easy",
+  MEDIUM = "medium",
+  HARD = "hard",
+}
+
+export type QuestionsState = Question & { answers: string[] };
+
+export const fetchQuizQuestions = async (amount: number, difficulty: Difficulty): Promise<QuestionsState[]> => {
+  const endpoint = `https://opentdb.com/api.php?amount=${amount}&difficulty=${difficulty}&type=multiple`;
+  const data = await (await fetch(endpoint)).json();
+  return data.results.map((question: Question) => ({
```

Comment:
should add a check here if the data or results is undefined from the API due to some issue at their end - 
`return data?.results?.map(`